fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -0,0 +1,221 @@
+        fdb.run(context -> {
+            FDBRecordStore.deleteStore(context, path);
+            return null;
+        });
+    }
+
+    private void rebuildIndexMetaData(final FDBRecordContext context, final String document, final Index index) {
+        Pair<FDBRecordStore, QueryPlanner> pair = LuceneIndexTestUtils.rebuildIndexMetaData(context, path, document, index, useRewritePlanner);
+        this.recordStore = pair.getLeft();
+        this.planner = pair.getRight();
+    }
+
+    private void disableIndex(Index index, String document) {
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, document, index);
+            recordStore.markIndexDisabled(index).join();
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void luceneOnlineIndexingTestSimple(boolean preventMergeDuringIndexing) {
+        Index index = SIMPLE_TEXT_SUFFIXES;
+        disableIndex(index, SIMPLE_DOC);
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            recordStore.saveRecord(createSimpleDocument(2222L, WAYLON + " who?", 1));
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            recordStore.saveRecord(createSimpleDocument(2222L, WAYLON + " who?", 1));
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setRecordStore(recordStore)
+                    .setIndex(index)
+                    .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                            .setPreventLuceneMergeDuringIndexing(preventMergeDuringIndexing))
+                    .build()) {
+                assertTrue(recordStore.isIndexDisabled(index));
+                indexBuilder.buildIndex(true);
+            }
+        }
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            assertTrue(recordStore.isIndexReadable(index));
+        }
+    }
+
+    @Test
+    void luceneOnlineIndexingTest1() {
+        luceneOnlineIndexingTestAny(QUERY_ONLY_SYNONYM_LUCENE_INDEX, COMPLEX_DOC, 7);
+    }
+
+    @Test
+    void luceneOnlineIndexingTest2() {
+        luceneOnlineIndexingTestAny(QUERY_ONLY_SYNONYM_LUCENE_INDEX, SIMPLE_DOC, 15);
```

Comment:
It seems like there should also be a test with multiple lucene indexes.

Reply:
Is there any other multi Lucene index test in the code that I can use as an example?

Reply:
I don't think so. I think this is the first test that does anything that might realistically be impacted by having more than one lucene index.
A simple approach would be to do something similar to what you have in: 
but add all grouping counts at once in a single test: `luceneOnlineIndexingTestGroupingKeys`
```
RecordMetaDataHook hook = metaDataBuilder -> {
    for (int i=0; i<3; i++) {
        metaDataBuilder.addIndex(new Index(
                                         "Map_with_auto_complete$entry-value-" + groupingCount,
                                         new GroupingKeyExpression(field("entry",
                                                                         KeyExpression.FanType.FanOut).nest(concat(LuceneIndexTest.keys)), groupingCount + 1),
                                         LuceneIndexTypes.LUCENE,
                                         ImmutableMap.of()));
    }
    metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
    TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
    metaDataBuilder.addIndex(MAP_DOC, index);
};
```

Reply:
✅ 

Added `luceneOnlineIndexingTestMulti`.